### PR TITLE
CinemaHall, Movie and MovieSession controllers implemented

### PR DESCRIPTION
**Pull request by hpavlo**

### CODE REVIEW

File: src/main/java/mate/academy/spring/controller/MovieController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    @Autowired
+    public MovieController(MovieService movieService, MovieMapper movieMapper) {
+        this.movieService = movieService;
+        this.movieMapper = movieMapper;
+    }
+
+    @PostMapping()
+    public MovieResponseDto add(@RequestBody MovieRequestDto movieRequestDto) {
+        Movie movie = movieMapper.toModel(movieRequestDto);
+        return movieMapper.toDto(movieService.add(movie));
```
Comment (YuriiHryniv) [inline src/main/java/mate/academy/spring/controller/MovieController.java:32]: I think you could return it in one single statement

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping
+    public MovieSessionResponseDto add(@RequestBody MovieSessionRequestDto movieSessionRequestDto) {
+        MovieSession movieSession = movieSessionMapper.toModel(movieSessionRequestDto);
+        return movieSessionMapper.toDto(movieSessionService.add(movieSession));
```
Comment (YuriiHryniv) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:39]: The same here

**Review by YuriiHryniv — APPROVED** [review]
LGTM

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/cinema-halls")
+public class CinemaHallController {
+    private final CinemaHallService cinemaHallService;
+    private final CinemaHallMapper cinemaHallMapper;
+
+    @Autowired
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:23]: Why do you need this annotation here?

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @PostMapping()
+    public CinemaHallResponseDto add(@RequestBody CinemaHallRequestDto cinemaHallRequestDto) {
+        CinemaHall cinemaHall = cinemaHallMapper.toModel(cinemaHallRequestDto);
+        return cinemaHallMapper.toDto(cinemaHallService.add(cinemaHall));
+    }
+
+    @GetMapping()
+    public List<CinemaHallResponseDto> get() {
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/spring/controller/CinemaHallController.java:37]:
```suggestion
    public List<CinemaHallResponseDto> getAll() {
```
get its usual name for returning some object by id

File: src/main/java/mate/academy/spring/controller/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+    }
+
+    @GetMapping("/available")
+    public List<MovieSessionResponseDto> get(@RequestParam Long movieId,
```
Comment (liliia-ponomarenko) [inline src/main/java/mate/academy/spring/controller/MovieSessionController.java:43]:
```suggestion
    public List<MovieSessionResponseDto> getAllByDate(@RequestParam Long movieId,
```

**Review by liliia-ponomarenko — CHANGES_REQUESTED** [review]
good job! Let’s  fix minor mistakes

File: src/main/java/mate/academy/spring/controller/CinemaHallController.java (hunk 3 above)
Comment (liliia-ponomarenko) [reply src/main/java/mate/academy/spring/controller/CinemaHallController.java:23]: As of Spring Framework 4.3, an @Autowired annotation on such a constructor is no longer necessary if the target bean only defines one constructor to begin with.
https://docs.spring.io/spring-framework/docs/4.3.x/spring-framework-reference/htmlsingle/#beans-autowired-annotation

Comment (liliia-ponomarenko) [reply src/main/java/mate/academy/spring/controller/CinemaHallController.java:23]: your choice  I just give you advice 😉

**Review by liliia-ponomarenko — APPROVED** [review]
great 😊

**Pull request closed without merging**